### PR TITLE
Use LIMS API for PHSKC Metadata Parse and Sample Matching

### PR DESCRIPTION
**Pull request by bencap**
Deprecates our use of the aliquoting sheets during the PHSKC clinical parsing routine in favor of making calls to the LIMS API in order to fetch the sample identifiers for records based on PHSKC CIDs. 

Adds a function that prepares LIMS API requests with HMAC signing. This function can be used by other routines as we remove all uses of the aliquoting sheets from our processes.

Must be merged alongside https://github.com/seattleflu/backoffice/pull/362

### CODE REVIEW

File: lib/seattleflu/id3c/cli/command/clinical.py [hunk 1]
```diff
@@ -456,36 +464,198 @@ def add_kp_manifest_data(df: pd.DataFrame, manifest_filenames: tuple, manifest_f
...
+    if we need to parse this file again.
+
+    Clinical records are parsed and transformed into suitable data for downstream
+    CID matching. PII is not removed in this function. Parsed data will later be joined
+    with barcode data present in the LIMS.
+
+    All clinical records (both newly and previously parsed data) are output to stdout
+    as newline-delimited JSON records. You will likely want to redirect stdout to a file.
+    """
+    parsed_clinical_records = pd.read_json(phskc_manifest_filename, orient='records', dtype={'inferred_symptomatic': 'str'}, lines=True)
```
Comment (davereinhart) [inline lib/seattleflu/id3c/cli/command/clinical.py:491]: Could add a check to make sure `phskc_manifest_filename` file exists. If it doesn't exist, there's an uncaught exception.

File: lib/seattleflu/id3c/cli/command/clinical.py [hunk 2]
```diff
@@ -456,36 +464,198 @@ def add_kp_manifest_data(df: pd.DataFrame, manifest_filenames: tuple, manifest_f
...
+            clinical_records = pd.read_excel(file, dtype={'inferred_symptomatic': 'str'})
+            clinical_records.columns = clinical_records.columns.str.lower()
+            clinical_records = trim_whitespace(clinical_records)
+        else:
+            LOG.debug(f'Skipped parsing of {relative_filename}, file has not been modified since last parse')
+            continue
+
+        if clinical_records.empty:
+            LOG.debug(f'Skipped parsing of {relative_filename}, file was empty')
+            continue
```
[the remark this answers is not in the document]
Comment (bencap) [reply lib/seattleflu/id3c/cli/command/clinical.py:520]: Good idea, I can add that in. I doubt it has happened but it would be good to check.

**Review by davereinhart — APPROVED** [review]
Nice work on this!

**Comment by bencap** [comment]
Deployed.

**Pull request merged**